models/repo_list.go

```diff
@@ -308,11 +321,15 @@ func accessibleRepositoryCondition(userID int64) builder.Cond {
 				builder.NotIn("`repository`.owner_id", builder.Select("id").From("`user`").Where(builder.Eq{"visibility": structs.VisibleTypePrivate}))),
 		),
 		// 2. Be able to see all repositories that we have access to
-		builder.In("`repository`.id", builder.Select("repo_id").
-			From("`access`").
-			Where(builder.And(
-				builder.Eq{"user_id": userID},
-				builder.Gt{"mode": int(AccessModeNone)}))),
+		builder.Or(
+			builder.In("`repository`.id", builder.Select("repo_id").
+				From("`access`").
+				Where(builder.And(
+					builder.Eq{"user_id": userID},
+					builder.Gt{"mode": int(AccessModeNone)}))),
+			builder.In("`repository`.id", builder.Select("id").
+				From("`repository`").
+				Where(builder.Eq{"owner_id": userID}))),
```

Comment:
@lunny Let me know if I was perhaps querying wrong, but I was unable to query private repos that the user owned (non-org) so I had to amend this function.  
It looks as though this was querying access, however I didn't see any data in that table for the repo owner (which makes sense, since they have full access)

Reply:
This logic seems to run when the `uid` is set in the search **and** `private` is true  
`uid` states `search only for repos that the user with the given id owns or contributes to`  
  
However, `private` states `include private repositories this user has access to (defaults to true)`

Which one takes precedent?   
The way my changes work (in order to show private template repos that the context user owns) is including all repos the user owns.  
The comment above the lines says `// 2. Be able to see all repositories that we have access to` so it made sense to change it accordingly.